Backpack-SwiftUI/CardButton/Classes/BPKSaveCardButton.swift

```diff
@@ -0,0 +1,142 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2024 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKSaveCardButton: View {
+    
+    private let animationDuration: TimeInterval = 0.2
+    private let maxScale: CGFloat = 1.25
+    @State private var scale: CGFloat = 1.0
+    @State private var isPressed: Bool = false
+    
+    let size: BPKCardButtonSize
+    let style: BPKCardButtonStyle
+    let checked: Bool
+    let accessibilityLabel: String
+    let action: () -> Void
+    
+    public init(
+        size: BPKCardButtonSize,
+        style: BPKCardButtonStyle,
+        checked: Bool,
+        accessibilityLabel: String,
+        action: @escaping () -> Void
+    ) {
+        self.size = size
+        self.style = style
+        self.checked = checked
+        self.accessibilityLabel = accessibilityLabel
+        self.action = action
+    }
+    
+    public var body: some View {
+        Button(action: action) {
+            BPKIconView(icon, size: size.iconSize)
+                .foregroundColor(color)
+                .scaleEffect(scale)
+        }
+        .buttonStyle(InternalSaveCardButtonStyle(onIsPressedchange: { isPressed = $0 }))
+        .frame(width: size.containerSize, height: size.containerSize)
+        .background(style.containerColor)
+        .clipShape(Circle())
+        .accessibilityLabel(accessibilityLabel)
```

Comment:
If the button is on/off we need a different a11y label. (Save hotel / Remove from saved) for example

Reply:
I don't think the button has to know the accessibilityLabel in both states. Because its state can only be changed by the caller. The caller should be responsible for providing the correct accessibilityLabel in each state.
Like the demo in CardButtonExampleView.swfit
```swift
private struct BPKSaveCardButtonInteractiveDemo: View {
    let size: Backpack_SwiftUI.BPKCardButtonSize
    let style: Backpack_SwiftUI.BPKCardButtonStyle
    @State var checked: Bool
    
    var body: some View {
        BPKSaveCardButton(
            size: size,
            style: style,
            checked: checked,
            accessibilityLabel: checked ? "remove from saved" : "save",
            action: {
                checked.toggle()
            }
        )
    }
}
```
WDYT?

Reply:
yes, apologies, This makes sense. 
Can you verify the SwiftUI card button works the same as the UIKit one?

Reply:
Yes, it works the same as the UIKit one after the last commit.